python/paddle/tensor/tensor.prototype.pyi

```diff
@@ -37,7 +37,24 @@ from paddle.base.dygraph.tensor_patch_methods import (
     TensorHookRemoveHelper,  # noqa: F401
 )
 
-class Tensor:
+# annotation: ${eager_param_base_begin}
+class EagerParamBase_(Protocol):
+    # annotation: ${eager_param_base_docstring}
+
+    # annotation: ${eager_param_base_attributes}
+
+    # annotation: ${eager_param_base_methods}
+    @property
+    def trainable(self) -> bool: ...
+    @trainable.setter
+    def trainable(self, trainable: bool) -> None: ...
+
+    # annotation: ${eager_param_base_alias}
```

Comment:
每个都有 Tensor 完整方法吗？会不会冲突？

Reply:
不会 ～

``` python
            if member_id in all_members:
                continue
```

`EagerParamBase_` 只会添加独有的方法 ～

部分 `tensor.pyi`

``` python
# annotation: ${eager_param_base_begin}
class EagerParamBase_(Protocol):
    # annotation: ${eager_param_base_docstring}
    r"""

        EagerParamBase is derived from Tensor( Which is the concept in Eager-Dygraph Mode).
        A EagerParamBase is a persistable Tensor, and will be updated by optimizers
        after each iteration.
        The training of a neural network is essentially the updating of
        its EagerParamBase.

        Relative to a general Tensor, a EagerParamBase has several its own
        member variables:

        Args:
            trainable(bool): True if the EagerParamBase need to be updated after
                iterations.
            optimize_attr(map): EagerParamBase attributes related with optimizing.
                Currently, it only contains 'learning_rate'.
                Default: {'learning_rate': 1.0}
            regularizer(WeightDecayRegularizer): The Regularizer which will
                be applied on the EagerParamBase. Default: None
            do_model_average(bool): True if the model average strategy will
                be applied on this EagerParamBase.
            need_clip (bool): Whether the parameter gradient need to be clipped
                in optimizer. Default is True.
        
    """


    # annotation: ${eager_param_base_attributes}

    # annotation: ${eager_param_base_methods}
    def __deepcopy__(self, memo):
        r"""
        Deep copy parameter, it will always performs Tensor copy.

        Examples:
            .. code-block:: python

                >>> import paddle
                >>> import copy
                >>> linear = paddle.nn.Linear(1, 3)
                >>> linear_copy = copy.deepcopy(linear)

                >>> print(linear.weight)
                >>> # doctest: +SKIP('it will be different')
                Parameter containing:
                Tensor(shape=[1, 3], dtype=float32, place=Place(cpu), stop_gradient=False,
                    [[-0.30929261, -0.90929240, -1.07851017]])

                >>> # doctest: -SKIP
                >>> print(linear_copy.weight)
                >>> # doctest: +SKIP('it will be different')
                Parameter containing:
                Tensor(shape=[1, 3], dtype=float32, place=Place(cpu), stop_gradient=False,
                    [[-0.30929261, -0.90929240, -1.07851017]])

        """
        ...

    def __init__(self, shape, dtype, **kwargs):
        r"""
        Initialize self.  See help(type(self)) for accurate signature.

        """
        ...

    def __repr__(self):
        r"""
        Convert a EagerParamBase object to a readable string.

        Returns(str): A readable string.

        Examples:
            .. code-block:: python

                >>> import paddle
                >>> linear = paddle.nn.Linear(3, 3)
                >>> print(linear.weight)
                >>> # doctest: +SKIP('it will be different')
                Parameter containing:
                Tensor(shape=[3, 3], dtype=float32, place=Place(cpu), stop_gradient=False,
                [[ 0.48948765,  0.05829060, -0.25524026],
                 [-0.70368278,  0.52986908, -0.68742192],
                 [-0.54217887,  0.48439729,  0.34082305]])

        """
        ...

    def from_tensor(self, **kwargs):
        ...

    def initialize(self):
        ...

    def set_init_func(self, obj):
        ...

    @property
    def trainable(self) -> bool: ...

    @trainable.setter
    def trainable(self, trainable: bool) -> None: ...

    # annotation: ${eager_param_base_alias}
    __str__ = __repr__

# annotation: ${eager_param_base_end}

# annotation: ${tensor_begin}
class Tensor_(Protocol):
    # annotation: ${tensor_docstring}
```